### ISSUE TITLE
Tests warning list VS13 Update4 x64

### ISSUE THREAD

**Issue by mathiask88** [rvagg/nan#255]
@kkoopa I saw you are on a long way to decimate the count of compile warnings in MSVC :wink:

So here is a full list of a x64 OS (AppVeyor runs on Server 2012 x86 I think)
```
Warning	1	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	asyncprogressworker
Warning	2	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	asyncprogressworker
Warning	3	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	asyncprogressworker
Warning	4	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	asyncworkererror
Warning	5	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	asyncworkererror
Warning	6	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	asyncworkererror
Warning	7	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	asyncprogressworker
Warning	8	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	asyncworkererror
Warning	9	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	asyncprogressworker
Warning	10	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	asyncworkererror
Warning	11	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	asyncworkererror
Warning	12	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	asyncworker
Warning	13	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	asyncworker
Warning	14	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	asyncworker
Warning	15	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	asyncprogressworker
Warning	16	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	asyncworker
Warning	17	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	asyncworker
Warning	18	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	bufferworkerpersistent
Warning	19	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	bufferworkerpersistent
Warning	20	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	bufferworkerpersistent
Warning	21	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	makecallback
Warning	22	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	makecallback
Warning	23	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	makecallback
Warning	24	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	asyncworker
Warning	25	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	bufferworkerpersistent
Warning	26	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	makecallback
Warning	27	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	gc
Warning	28	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	gc
Warning	29	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	gc
Warning	30	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	bufferworkerpersistent
Warning	31	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	makecallback
Warning	32	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	gc
Warning	33	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	makecallback
Warning	34	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	gc
Warning	35	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	isolatedata
Warning	36	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	isolatedata
Warning	37	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	isolatedata
Warning	38	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	bufferworkerpersistent
Warning	39	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	gc
Warning	40	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	isolatedata
Warning	41	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	isolatedata
Warning	42	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	morenews
Warning	43	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	morenews
Warning	44	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	morenews
Warning	45	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	morenews
Warning	46	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	isolatedata
Warning	47	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	morenews
Warning	48	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	morenews
Warning	49	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	nannew
Warning	50	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	nannew
Warning	51	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	nannew
Warning	52	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	news
Warning	53	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	nannew
Warning	54	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	news
Warning	55	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	news
Warning	56	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	nannew
Warning	57	warning C4800: 'gboolean' : forcing value to bool 'true' or 'false' (performance warning)	c:\users\mathias\documents\github\nan\nan_new.h	185	1	nannew
Warning	58	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	news
Warning	59	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	nannew
Warning	60	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	news
Warning	61	warning C4800: 'int' : forcing value to bool 'true' or 'false' (performance warning)	c:\users\mathias\documents\github\nan\nan_new.h	185	1	news
Warning	62	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	optionvalues
Warning	63	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	optionvalues
Warning	64	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	optionvalues
Warning	65	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	news
Warning	66	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	optionvalues
Warning	67	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	optionvalues
Warning	68	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	optionvalues
Warning	69	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data (..\cpp\multifile1.cpp)	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	multifile
Warning	70	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data (..\cpp\multifile1.cpp)	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	multifile
Warning	71	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data (..\cpp\multifile2.cpp)	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	multifile
Warning	72	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data (..\cpp\multifile2.cpp)	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	multifile
Warning	73	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data (..\cpp\multifile1.cpp)	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	multifile
Warning	74	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data (..\cpp\multifile2.cpp)	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	multifile
Warning	75	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc (..\cpp\multifile1.cpp)	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	multifile
Warning	76	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc (..\cpp\multifile2.cpp)	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	multifile
Warning	77	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc (..\cpp\multifile1.cpp)	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	multifile
Warning	78	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc (..\cpp\multifile2.cpp)	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	multifile
Warning	79	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)' (..\cpp\multifile2.cpp)	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	multifile
Warning	80	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)' (..\cpp\multifile1.cpp)	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	multifile
Warning	81	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	persistent
Warning	82	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	persistent
Warning	83	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	persistent
Warning	84	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	returnnull
Warning	85	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	returnnull
Warning	86	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	returnnull
Warning	87	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	returnemptystring
Warning	88	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	returnemptystring
Warning	89	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	returnemptystring
Warning	90	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	persistent
Warning	91	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	returnnull
Warning	92	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	returnnull
Warning	93	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	returnemptystring
Warning	94	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	persistent
Warning	95	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	returnundefined
Warning	96	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	returnundefined
Warning	97	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	returnemptystring
Warning	98	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	returnundefined
Warning	99	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	returnnull
Warning	100	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	persistent
Warning	101	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	returnemptystring
Warning	102	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	returnundefined
Warning	103	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	returnundefined
Warning	104	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	returnvalue
Warning	105	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	returnvalue
Warning	106	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	returnvalue
Warning	107	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	returnundefined
Warning	108	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	returnvalue
Warning	109	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	returnvalue
Warning	110	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	settemplate
Warning	111	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	settemplate
Warning	112	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	settemplate
Warning	113	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	returnvalue
Warning	114	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	settemplate
Warning	115	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	settemplate
Warning	116	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	settergetter
Warning	117	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	settergetter
Warning	118	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	settergetter
Warning	119	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	settemplate
Warning	120	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	settergetter
Warning	121	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	settergetter
Warning	122	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	184	1	settergetter
Warning	123	warning C4506: no definition for inline function 'v8::Persistent<v8::FunctionTemplate> v8::Persistent<v8::FunctionTemplate>::New(v8::Handle<v8::FunctionTemplate>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	184	1	settergetter
Warning	124	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	strings
Warning	125	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	strings
Warning	126	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	strings
Warning	127	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	strings
Warning	128	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	symbols
Warning	129	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	symbols
Warning	130	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	strings
Warning	131	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	symbols
Warning	132	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	strings
Warning	133	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	symbols
Warning	134	warning C4506: no definition for inline function 'v8::Persistent<v8::FunctionTemplate> v8::Persistent<v8::FunctionTemplate>::New(v8::Handle<v8::FunctionTemplate>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	strings
Warning	135	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	symbols
Warning	136	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	symbols
Warning	137	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	161	1	weak
Warning	138	warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	170	1	weak
Warning	139	warning C4267: 'argument' : conversion from 'size_t' to 'int', possible loss of data	c:\users\mathias\documents\github\nan\nan_implementation_pre_12_inl.h	188	1	weak
Warning	140	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1107	1	weak
Warning	141	warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\vector	1590	1	weak
Warning	142	warning C4506: no definition for inline function 'v8::Persistent<v8::Object> v8::Persistent<v8::Object>::New(v8::Handle<v8::Object>)'	C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include\v8.h	179	1	weak
```

All Options:
```
/GS /GL /W3 /wd"4251" /Gy /Zc:wchar_t /I"..\.." /I"C:\Users\Mathias\.node-gyp\0.10.35\src" /I"C:\Users\Mathias\.node-gyp\0.10.35\deps\uv\include" /I"C:\Users\Mathias\.node-gyp\0.10.35\deps\v8\include" /Zi /Gm- /Ox /Ob2 /Fd"Release\obj\asyncprogressworker\vc120.pdb" /fp:precise /D "WIN32" /D "_CRT_SECURE_NO_DEPRECATE" /D "_CRT_NONSTDC_NO_DEPRECATE" /D "BUILDING_V8_SHARED=1" /D "BUILDING_UV_SHARED=1" /D "BUILDING_NODE_EXTENSION" /D "_WINDLL" /errorReport:prompt /GF /WX- /Zc:forScope /GR- /Gd /Oy /Oi /MT /Fa"Release\obj\asyncprogressworker\" /nologo /Fo"Release\obj\asyncprogressworker\" /Ot /Fp"Release\obj\asyncprogressworker\asyncprogressworker.pch" 
```

This are all options passed to the compiler from VS for example "asyncprogressworker" test. The others are analogous.

**Comment by kkoopa** [comment]
Greatly appreciated. I intend to get rid of all warnings originating from our code. Unfortunately nothing can be done about those from v8 and node.

**Comment by kkoopa** [comment]
Should not be much left after #256.

**Comment by mathiask88** [comment]
What about all the C4267 size_t to int?

**Comment by kkoopa** [comment]
Should have been fixed in #247

**Comment by mathiask88** [comment]
Yes everything is fixed if #256 gets merged and #254 is updated. Very nice :)
